src/sentry/api/endpoints/organization_events_timeseries.py

```diff
@@ -0,0 +1,371 @@
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get_features(
+        self, organization: Organization, request: Request
+    ) -> Mapping[str, bool | None]:
+        feature_names = [
+            "organizations:performance-use-metrics",
+            "organizations:dashboards-mep",
+            "organizations:mep-rollout-flag",
+        ]
+        batch_features = features.batch_has(
+            feature_names,
+            organization=organization,
+            actor=request.user,
+        )
+        return (
+            batch_features.get(f"organization:{organization.id}", {})
+            if batch_features is not None
+            else {
+                feature_name: features.has(
+                    feature_name, organization=organization, actor=request.user
+                )
+                for feature_name in feature_names
+            }
+        )
+
+    def get_request_querysource(self, request: Request, referrer: str) -> QuerySource:
+        if referrer in SENTRY_BACKEND_REFERRERS:
+            return QuerySource.SENTRY_BACKEND
+        else:
+            return self.get_request_source(request)
+
+    def get_top_events(self, request: Request) -> int:
+        if "topEvents" in request.GET:
+            try:
+                top_events = int(request.GET.get("topEvents", 0))
+            except ValueError:
+                raise ParseError(detail="topEvents must be an integer")
+            if top_events > MAX_TOP_EVENTS:
+                raise ParseError(detail=f"Can only get up to {MAX_TOP_EVENTS} top events")
+            elif top_events <= 0:
+                raise ParseError(detail="topEvents needs to be at least 1")
+
+            return top_events
+        else:
+            return 0
+
+    def get_comparison_delta(self, request: Request) -> timedelta | None:
+        if "comparisonDelta" in request.GET:
+            try:
+                return timedelta(seconds=int(request.GET["comparisonDelta"]))
+            except ValueError:
+                raise ParseError(detail="comparisonDelta must be an integer")
+        else:
+            return None
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        with sentry_sdk.start_span(op="discover.endpoint", name="filter_params") as span:
+            span.set_data("organization", organization)
+
+            top_events = self.get_top_events(request)
+            comparison_delta = self.get_comparison_delta(request)
+
+            dataset = self.get_dataset(request)
+            # Add more here until top events is supported on all the datasets
+            if top_events > 0:
+                if dataset not in TOP_EVENTS_DATASETS:
+                    raise ParseError(detail=f"{dataset} doesn't support topEvents yet")
+
+            metrics_enhanced = dataset in {metrics_performance, metrics_enhanced_performance}
+            use_rpc = dataset in {spans_indexed, ourlogs, uptime_checks}
```

Comment:
do we need to include `spans_eap` in this set?

Reply:
Good reminder, going to remove `spans_eap` as a potential dataset in this endpoint, and move it to the future endstate all the endpoints should be on anyways where `dataset=spans` actually means `spans_rpc`